File: src/desktop/ui/src/Actions/Models/ActionErrorKind.ts
export const actionErrorKinds = [
  "operation",
  "transport",
  "protocol",
  "unexpected",
] as const;

export type ActionErrorKind = (typeof actionErrorKinds)[number];

export function isActionErrorKind(value: unknown): value is ActionErrorKind {
  return actionErrorKinds.some((kind) => kind === value);
}

File: src/desktop/src-tauri/src/engine_protocol/models/action_error_kind.rs
use serde::{Deserialize, Serialize};

#[derive(Clone, Copy, Debug, Deserialize, PartialEq, Eq, Serialize)]
#[serde(rename_all = "camelCase")]
pub enum ActionErrorKind {
    Operation,
    Transport,
    Protocol,
    Unexpected,
}
